src/components/side_nav/side_nav_item.tsx

```diff
@@ -171,18 +174,31 @@ export function EuiSideNavItem<
   truncate = true,
   emphasize,
   buttonClassName,
+  childrenOnly,
   ...rest
 }: EuiSideNavItemProps<T>) {
   const isHrefValid = !_href || validateHref(_href);
   const href = isHrefValid ? _href : '';
-  let childItems;
+  const isClickable = onClick || href;
+
+  // Forcing accordion style item if not linked, but has children
+  const [itemIsOpen, setItemIsOpen] = useState(isOpen);
+  useEffect(() => {
+    if (isOpen) setItemIsOpen(true);
+    return () => {
+      setItemIsOpen(false);
+    };
```

Comment:
I believe this condenses down to:

```suggestion
    setItemIsOpen(isOpen);
```

As the code right now, when `isOpen` changes:
* sets itemIsOpen to false (via the returned function, as the previous effect is thrown away)
* if `isOpen` is true, set itemIsOpen to true

Reply:
So you don't always need a `return` to clear the `useEffect`?

Reply:
It depends (of course it does). Hopefully my pre-coffee self writes this up well enough, if it's still unclear please let me know!

The returned function will be executed when the component is unmounted or if an item in the useEffect's dependency array changes, before the next useEffect is setup. The returned function is generally for interacting with non-React lifecycle items like the DOM or some API.

In this component's case, the only interaction is with a React lifecycle through the state setter. That state value is unimportant when the component is unmounting, so it doesn't need to clear at that point, and the useEffect function will fire again when `isOpen` changes, giving the opportunity to directly set the state to the new value.

Reply:
Thanks for the explanation! I will probably continue to be a bit confused by this, but it's the type of feedback I really appreciate in these PR's so thank you!